lib/widget/button.dart

```diff
@@ -168,7 +170,7 @@ class ButtonState extends WidgetState<Button> {
 
     // execute the onTap action
     if (widget._controller.onTap != null) {
-      ScreenController().executeAction(context, widget._controller.onTap!);
+      ScreenController().executeAction(context, widget._controller.onTap!,event: EnsembleEvent(widget));
```

Comment:
not a fan of having to explicitly set the event - I'd forget. I guess we can refactor later to dispatch the Action as an action instead of calling a Singleton. Widget will then automatically inject itself in

Reply:
@vusters note that event is NOT required. It must not be required because an action can be triggered without an event as well. You should be able to trigger an action from the code or trigger an event from the code. Event should be there only when the action is being triggered as a result of an event. So we can't just always add it.